File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
         int maxdiff = INT_MIN,minprice = INT_MAX;
        for(int price:prices){
             minprice = min(price,minprice);
             maxdiff = max(maxdiff,price-minprice);
        }
        return maxdiff;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
       int last = prices[0], ans = 0;
        for(int i=1;i<prices.length;i++){
          if(last < prices[i]){
            ans = Math.max(ans, prices[i] - last);
          }
          last = Math.min(last, prices[i]);
        }

        return ans;
    }
}